sdk/trace/span.go

```diff
@@ -347,17 +347,24 @@ func (s *recordingSpan) End(options ...trace.SpanEndOption) {
 		return
 	}
 
+	config := trace.NewSpanEndConfig(options...)
```

Comment:
What other impact can moving the processing of config options before the `IsRecording()` short-circuit have?

Reply:
Hi @Aneurysm9, we only have two kinds of `SpanEndOption`, which are `trace.WithStackTrace` and `trace.WithTimestamp`. And the `config := trace.NewSpanEndConfig(options...)` is allocating on the stack, no GC impact. Therefore the impact will be looping at most two iterations on the options.

Reply:
Does anyone understand the comment below this, "now that we have an end time and see if we need to do any more processing."

I can't see why we'd compute the timestamp before the short-circuit.

Reply:
> Does anyone understand the comment below this, "now that we have an end time and see if we need to do any more processing."
> 
> I can't see why we'd compute the timestamp before the short-circuit.

Because if we didn't the end time would be less accurate.